### PR TITLE
[nobug] Refactor Nimbus tests to use correct `*OnThisThread()` functions

### PR DESCRIPTION
**Pull request by travis79**

### CODE REVIEW

**Comment by travis79** [comment]
@jhugman Can I get you to review this when you get a chance?  Thanks!

File: components/service/nimbus/src/test/java/mozilla/components/service/nimbus/NimbusTest.kt [hunk 1]
```diff
@@ -143,24 +133,11 @@ class NimbusTest {
     }
 
     @Test
-    @Ignore("https://github.com/mozilla-mobile/android-components/issues/9635")
+//    @Ignore("https://github.com/mozilla-mobile/android-components/issues/9635")
```
Comment (jhugman) [inline components/service/nimbus/src/test/java/mozilla/components/service/nimbus/NimbusTest.kt:136]: Nit: remove dead code.

File: components/service/nimbus/src/test/java/mozilla/components/service/nimbus/NimbusTest.kt [hunk 2]
```diff
@@ -190,22 +167,10 @@ class NimbusTest {
...
-
-        assertTrue("Timeout waiting on applying updates", waitCount > 0)
+        nimbus.applyPendingExperimentsOnThisThread()
 
         // Record the exposure event in Glean
-        nimbus.getExperimentBranch("test-experiment")
-        // Test is failing here because we are recording exposure events on another thread,
-        // and we aren't exposing the dbScope to mock it, or use it to run tests on.
-        // https://github.com/mozilla-mobile/android-components/issues/9635
+        nimbus.recordExposureOnThisThread("test-experiment")
```
Comment (jhugman) [inline components/service/nimbus/src/test/java/mozilla/components/service/nimbus/NimbusTest.kt:173]: 😻

**Review by jhugman — APPROVED** [review]
Great!

File: components/service/nimbus/src/test/java/mozilla/components/service/nimbus/NimbusTest.kt (hunk 1 above)
Comment (travis79) [reply components/service/nimbus/src/test/java/mozilla/components/service/nimbus/NimbusTest.kt:136]: Gah, how did I miss this 🤦    Good eye!

**Comment by travis79** [comment]
@jhugman would you care to re-review, now that you have been added to the AST team so that this can be landed when ready?  Thanks!

**Review by jhugman — APPROVED** [review]
LGTM!

**Pull request merged**